src/bindings/python/src/pyopenvino/core/tensor.cpp

```diff
@@ -177,11 +229,12 @@ void regclass_Tensor(py::module m) {
     cls.def_property_readonly(
         "data",
         [](ov::Tensor& self) {
-            return py::array(Common::ov_type_to_dtype().at(self.get_element_type()),
-                             self.get_shape(),
-                             self.get_strides(),
-                             self.data(),
-                             py::cast(self));
+            auto ov_type = self.get_element_type();
+            auto dtype = Common::ov_type_to_dtype().at(ov_type);
+            if (ov_type.bitwidth() < 8) {
+                return py::array(dtype, self.get_byte_size(), self.data(), py::cast(self));
```

Comment:
Yes, it can't have the same shape as Tensor, because Tensor shape is in the element types, which bitwidth is less then 8, so the only way here is to return it as linear numpy array.

Reply:
Can you add a note about it to the docs string below? It seems really important to note this.